### ISSUE TITLE
Assertion `M->getDataLayout() == getDataLayout() && "DataLayout Mismatch"' failed

### ISSUE THREAD

**Issue by JayceCao** [iovisor/bcc#2175]
* os: `Gentoo`
* kernel version: `inux 4.14.83-gentoo #1 SMP Sat Nov 24 18:34:29 CST 2018 x86_64 Intel(R) Core(TM) i7-4710HQ CPU @ 2.50GHz GenuineIntel GNU/Linux`
* llvm version: `6.0.1`
* llvm build settings: `USE="debug libffi ncurses -doc -gold -libedit -test -xar -xml" ABI_X86="(64) -32 (-x32)" LLVM_TARGETS="AArch64 ARM BPF Mips NVPTX (X86) -AMDGPU -Hexagon -Lanai -MSP430 -PowerPC -Sparc -SystemZ -XCore"`
* bcc version: `0.7.0`
* bcc build settings: `USE="" ABI_X86="(64)" PYTHON_SINGLE_TARGET="python3_6 -python2_7 -python3_4 -python3_5 -python3_7" PYTHON_TARGETS="python2_7 python3_6 -python3_4 -python3_5 -python3_7"`

* How to reproduce: execute `/usr/share/bcc/tools/opensnoop`, then it output:
```
ython3.6: /var/tmp/portage/sys-devel/llvm-6.0.1/work/llvm-6.0.1.src/lib/ExecutionEngine/MCJIT/MCJIT.cpp:199: virtual void llvm::MCJIT::generateCodeForModule(llvm::Module*): Assertion `M->getDataLayout() == getDataLayout() && "DataLayout Mismatch"' failed.
[1]    29735 abort (core dumped)  /usr/share/bcc/tools/opensnoop
```
So it's a bug or my mistake of installation? Thanks

**Comment by yonghong-song** [comment]
It is a bug in bcc. When the llvm is built with assertion on, the bug will show up. The bug is fixed by the following commit:
```
commit 166fba5734088347aa4dd3494ceddf01a48b28ab
Author: yonghong-song <ys114321@gmail.com>
Date:   Wed Nov 21 08:30:42 2018 -0800

    set proper layout for the module before finalizing the object (#2054)
    
    Otherwise, if llvm is built with assertion on,
    we will hit the following error for a simple trace.py script like
      -bash-4.4$ sudo ./trace.py 'SyS_nanosleep'
      python: /home/yhs/work/llvm/lib/ExecutionEngine/MCJIT/MCJIT.cpp:204:
              virtual void llvm::MCJIT::generateCodeForModule(llvm::Module*):
              Assertion `M->getDataLayout() == getDataLayout() && "DataLayout Mismatch"'
              failed.
      Aborted (core dumped)
      -bash-4.4$
    
    Signed-off-by: Yonghong Song <yhs@fb.com>
```
The bug is fix in bcc 0.8.0.

**Comment by JayceCao** [comment]
It might be. Upgrading to v0.8.0 can solve the problem.

> It is a bug in bcc. When the llvm is built with assertion on, the bug will show up. The bug is fixed by the following commit:
> 
> ```
> commit 166fba5734088347aa4dd3494ceddf01a48b28ab
> Author: yonghong-song <ys114321@gmail.com>
> Date:   Wed Nov 21 08:30:42 2018 -0800
> 
>     set proper layout for the module before finalizing the object (#2054)
>     
>     Otherwise, if llvm is built with assertion on,
>     we will hit the following error for a simple trace.py script like
>       -bash-4.4$ sudo ./trace.py 'SyS_nanosleep'
>       python: /home/yhs/work/llvm/lib/ExecutionEngine/MCJIT/MCJIT.cpp:204:
>               virtual void llvm::MCJIT::generateCodeForModule(llvm::Module*):
>               Assertion `M->getDataLayout() == getDataLayout() && "DataLayout Mismatch"'
>               failed.
>       Aborted (core dumped)
>       -bash-4.4$
>     
>     Signed-off-by: Yonghong Song <yhs@fb.com>
> ```
> The bug is fix in bcc 0.8.0.
